aptos-move/aptos-gas/src/misc.rs

```diff
@@ -166,27 +168,49 @@ impl<'a> ValueVisitor for AbstractValueSizeVisitor<'a> {
 
     #[inline]
     fn visit_vec_u8(&mut self, _depth: usize, vals: &[u8]) {
-        self.size += self.params.per_u8_packed * NumArgs::new(vals.len() as u64);
+        let mut size = self.params.per_u8_packed * NumArgs::new(vals.len() as u64);
+        if self.feature_version >= 2 {
+            size += self.params.vector;
+        }
+        self.size += size;
     }
 
     #[inline]
     fn visit_vec_u64(&mut self, _depth: usize, vals: &[u64]) {
-        self.size += self.params.per_u64_packed * NumArgs::new(vals.len() as u64);
+        let mut size = self.params.per_u64_packed * NumArgs::new(vals.len() as u64);
+        if self.feature_version >= 2 {
+            size += self.params.vector;
+        }
+        self.size += size;
     }
 
     #[inline]
     fn visit_vec_u128(&mut self, _depth: usize, vals: &[u128]) {
-        self.size += self.params.per_u128_packed * NumArgs::new(vals.len() as u64);
+        let mut size = self.params.per_u128_packed * NumArgs::new(vals.len() as u64);
+        if self.feature_version >= 2 {
+            size += self.params.vector;
+        }
+        self.size += size;
     }
 
     #[inline]
     fn visit_vec_bool(&mut self, _depth: usize, vals: &[bool]) {
-        self.size += self.params.per_bool_packed * NumArgs::new(vals.len() as u64);
+        let mut size =
+            self.params.vector + self.params.per_bool_packed * NumArgs::new(vals.len() as u64);
+        if self.feature_version >= 2 {
+            size += self.params.vector;
+        }
+        self.size += size;
     }
 
     #[inline]
     fn visit_vec_address(&mut self, _depth: usize, vals: &[AccountAddress]) {
-        self.size += self.params.per_address_packed * NumArgs::new(vals.len() as u64);
+        let mut size =
+            self.params.vector + self.params.per_address_packed * NumArgs::new(vals.len() as u64);
+        if self.feature_version >= 2 {
+            size += self.params.vector;
+        }
+        self.size += size;
```

Comment:
this isn't correct. You're adding in params.vector twice.

why didn't you just do

```
        self.size += self.params.per_address_packed * NumArgs::new(vals.len() as u64);
        if self.feature_version >= 2 {
            self.size += self.params.vector;
        }
```

Reply:
this is consistent with each of these chunks of code

Reply:
No `self.params.vector` is the size of the vector excluding the elements, which I forgot to add in the previous version so that's why I'm adding it back right now

Reply:
1. The previous code did not have `self.params.vector`
2. You are now adding `self.params.vector` twice in the case of feature 2
3. The written code of creating a new size value and adding to it is not very intuitive versus the code written

Reply:
Oh my bad... Will get it fixed